sdk/src/test/java/io/opentelemetry/sdk/trace/TestUtils.java

```diff
@@ -78,4 +80,21 @@ public static SpanData makeBasicSpan() {
         .setEndTimestamp(Timestamp.create(200, 200))
         .build();
   }
+
+  /**
+   * Create a very basic SpanData instance, suitable for testing. It has the bare minimum viable
+   * data.
+   *
+   * @return A SpanData instance.
+   */
+  public static Span.Builder startSpanWithSampler(
+      TracerSdk tracerSdk, String spanName, Sampler sampler) {
+    TraceConfig originalConfig = tracerSdk.getActiveTraceConfig();
+    tracerSdk.updateActiveTraceConfig(originalConfig.toBuilder().setSampler(sampler).build());
+    try {
+      return tracerSdk.spanBuilder(spanName);
+    } finally {
+      tracerSdk.updateActiveTraceConfig(originalConfig);
```

Comment:
It makes me uncomfortable that the test code needs to know about the internal details of how the span builder makes sampling decisions. I personally found it surprising that we could set the configuration back before the span has actually been built, and the span ends up with the correct sampling decision.

Reply:
I do agree with you and we may want to revisit this. I think this PR is important to make progress in order for us to be compatible with the specs.

Reply:
https://github.com/open-telemetry/opentelemetry-java/issues/629